### PR TITLE
Backport SCP-2417 and SCP-3619

### PR DESCRIPTION
**Pull request by bezirg**
I cherrypicked SCP-2417 and then applied manually ontop some changes of https://github.com/input-output-hk/plutus/commit/bd97960ecafe1be3647c9634021d5fe355b85d54 and https://github.com/input-output-hk/plutus/commit/32268d3d6fef17d2888ea204272ad90605ce5266 .

I manually added the new values for serialiseData into the old builtincostmodel.json

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

**Review by kwxm — APPROVED** [review]
Looks OK to me.

File: doc/reference/builtin-parameters.csv [hunk 1]
```diff
@@ -35,7 +35,7 @@ decodeUtf8,decodeUtf8-memory-arguments-slope,Linear model coefficient for the me
 divideInteger,divideInteger-cpu-arguments-constant,Constant CPU cost (argument sizes above diagonal)
 divideInteger,divideInteger-cpu-arguments-model-arguments-intercept,Linear model intercept for the CPU calculation (argument sizes on or below diagonal)
 divideInteger,divideInteger-cpu-arguments-model-arguments-slope,Linear model coefficient for the CPU calculation (argument sizes on or below diagonal)
-divideInteger,divideInteger-memory-arguments-intercept,Linear model intercept for the memory calculation (argument sizes on or below diagonal) 
+divideInteger,divideInteger-memory-arguments-intercept,Linear model intercept for the memory calculation (argument sizes on or below diagonal)
```
[the remark this answers is not in the document]
Comment (bezirg) [reply doc/reference/builtin-parameters.csv:38]: no idea, I was hoping you are going to tell me... :D

**Comment by michaelpj** [comment]
IMO simplest to exactly do cherry picks and new commits, so I'd slightly rather this was 3 commits. And don't worry about ignoring the R code.

**Comment by bezirg** [comment]
> IMO simplest to exactly do cherry picks and new commits, so I'd slightly rather this was 3 commits. And don't worry about ignoring the R code.

There are in-between commits that mess up the cherrypicking, e.g. #4468 , so instead of fighting the `diff` tool, i just handpicked the changes.

**Comment by michaelpj** [comment]
Okay :+1:

**Pull request merged**